### ISSUE TITLE
system size attibute L cause index() to fail for spinful_fermion_basis_general

### ISSUE THREAD

**Issue by anton-buyskikh** [weinbe58/QuSpin#243]
I use
```bash
quspin 0.3.2
numpy 1.16.4
```

The following code
```python
from quspin.basis import spinful_fermion_basis_general
basis = spinful_fermion_basis_general(5, Nf=(1, 1))
print(basis.index('10000', '10000'))
```
produces the following error
```bash
Traceback (most recent call last):
  File "error_large_L.py", line 3, in <module>
    print(basis.index('10000', '10000'))
  File "/home/anton/anaconda3/lib/python3.7/site-packages/quspin/basis/basis_general/fermion.py", line 582, in index
    s = down_state + (up_state << self.L)
AttributeError: 'spinful_fermion_basis_general' object has no attribute 'L'
```

I checked that `spinful_fermion_basis_1d` works fine. It seems that the issue is due to the fact that `*_basis_1d` have `L` and `*_basis_general` have `N` as the system size attributes, which probably means this issue should be extended to spins and bosons.

It seems to me that  `*_basis_1d` should be a child class of `*_basis_general` on the top of which the symmetries are built, which is not the case at the moment. I might be wrong about the last statement as I haven't investigated the code in great detail.

**Comment by weinbe58** [comment]
Well it turns out that there is a parent class for `*_basis_1d` but it is not `*_basis_general` it is something called `lattice_basis` but that's not that important, only in the sense that it does some basic operations that every lattice basis does. 

Every basis should have attribute `N` but only the 1D basis classes have an attribute `L`.

**Comment by mgbukov** [comment]
this was a leftover bug. 

@anton-buyskikh if it annoys you until the new release, just replace `self.L` by `(self.N//2)` in line 593 of `quspin/basis/basis_general/fermion.py`. 

I fixed this in `user_basis` in #245 . We'll close this issue with the new official release.

**Comment by anton-buyskikh** [comment]
Thank you, I can wait for the new release.
